src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/runtime/Threads.java

```diff
@@ -227,7 +228,24 @@ public JavaThread owningThreadFromMonitor(Address o) {
     }
 
     public JavaThread owningThreadFromMonitor(ObjectMonitor monitor) {
-        return owningThreadFromMonitor(monitor.owner());
+        if (VM.getVM().getCommandLineFlag("LockingMode").getInt() == 2) {
```

Comment:
Please put a comment after that literal '2':
```
        if (VM.getVM().getCommandLineFlag("LockingMode").getInt() == 2 /* LM_LIGHTWEIGHT */) {
```

Reply:
You could add the LM_LEGACY, LM_LIGHTWEIGHT literals to vmStructs.cpp and compare with them.